### PR TITLE
Add serializer for `akka.NotUsed` (#24599)

### LINKED ISSUE #24599: Missing serializer for `akka.NotUsed`
Related issue: https://github.com/akka/akka/issues/20507

### PR DESCRIPTION
**Pull request by franbh**
Fixes #24599

### CODE REVIEW

File: akka-actor/src/main/scala/akka/NotUsed.scala [hunk 1]
```diff
@@ -9,7 +9,7 @@ package akka
  * used from the other language. An example use-case is the materialized value of an Akka Stream for cases
  * where no result shall be returned from materialization.
  */
-sealed abstract class NotUsed
+sealed abstract class NotUsed extends Serializable
```
Comment (johanandren) [inline akka-actor/src/main/scala/akka/NotUsed.scala]: No need to make it Java-serializable

**Review by johanandren — APPROVED** [review]
LGTM with one comment

File: akka-actor/src/main/scala/akka/NotUsed.scala (hunk 1 above)
Comment (franbh) [reply akka-actor/src/main/scala/akka/NotUsed.scala]: Perfect, I'll fix it right away

**Comment by ktoso** [comment]
OK TO TEST

File: akka-remote/src/main/resources/reference.conf [hunk 2]
```diff
@@ -52,9 +52,10 @@ akka {
 
 
       # The following are handled by the MiscMessageSerializer, but they are not enabled for
-      # compatibility reasons (it was added in Akka 2.5.[8,9]). Enable them by adding:
+      # compatibility reasons (it was added in Akka 2.5.[8,9,10]). Enable them by adding:
```
Comment (patriknw) [inline akka-remote/src/main/resources/reference.conf]: 12 will be the next one, including this pr

**Review by patriknw — APPROVED** [review]
LGTM, with one nitpick

**Review by jrudolph — APPROVED** [review]
LGTM

A bit useless that we have different runtime objects for `NotUsed`, `Done` and `()`, but that's not your fault ;)

**Comment by jrudolph** [comment]
Fixed that version number in the config comment.

**Comment by jrudolph** [comment]
Thanks a lot, @franbh.

**Pull request merged**